### PR TITLE
hidapi 0.8.0-rc1 patched

### PR DESCRIPTION
**Pull request by rbarazzutti**
Fix for the HidApi IOHidManager back-end. hidapi 0.8.0-rc1 contains a bug in the implementation of hid_get_feature_report for the Mac platform. This bug prevents many uses of that library.

### CODE REVIEW

File: Library/Formula/hidapi.rb [hunk 1]
```diff
@@ -4,6 +4,20 @@ class Hidapi < Formula
   homepage 'https://github.com/signal11/hidapi'
   url 'https://github.com/signal11/hidapi/archive/hidapi-0.8.0-rc1.tar.gz'
   sha1 '5e72a4c7add8b85c8abcdd360ab8b1e1421da468'
+  
+  ## Patch - April 8th, 2015 - Raphaël P. Barazzutti
```
Comment (DomT4) [inline Library/Formula/hidapi.rb]: It's not hugely useful to other formula updaters/users to see who submitted the patch upstream. We don't tend to include author names. Up to the maintainers though.

File: Library/Formula/hidapi.rb [hunk 2]
```diff
@@ -4,6 +4,20 @@ class Hidapi < Formula
...
+
+  # This patch addresses a bug discovered in the HidApi IOHidManager back-end 
+  # that is being used with Macs.
+  # The bug was dramatically changing the behaviour of the function 
+  # "hid_get_feature_report". As a consequence, many applications working
+  # with HidApi were not behaving correctly on OSX.
+  #
```
Comment (DomT4) [inline Library/Formula/hidapi.rb]: Kill this blank comment.

File: Library/Formula/hidapi.rb [hunk 3]
```diff
@@ -4,6 +4,20 @@ class Hidapi < Formula
...
+
+  # This patch addresses a bug discovered in the HidApi IOHidManager back-end 
+  # that is being used with Macs.
+  # The bug was dramatically changing the behaviour of the function 
+  # "hid_get_feature_report". As a consequence, many applications working
+  # with HidApi were not behaving correctly on OSX.
+  # 
+  # pull request on Hidapi's repo: https://github.com/signal11/hidapi/pull/219
+  patch do
+    url 'https://patch-diff.githubusercontent.com/raw/signal11/hidapi/pull/219.patch'
```
Comment (DomT4) [inline Library/Formula/hidapi.rb]: Double quotes please.

File: Library/Formula/hidapi.rb [hunk 4]
```diff
@@ -4,6 +4,20 @@ class Hidapi < Formula
...
+    sha1 '087560554b232b7051c229eb14aa79adc189be22'
```
Comment (DomT4) [inline Library/Formula/hidapi.rb]: And SHA256 would be awesome :wink:

File: Library/Formula/hidapi.rb (hunk 3 above)
Comment (DomT4) [inline Library/Formula/hidapi.rb]: Oh, should be a ` .diff ` link rather than a `.patch` link as well.

File: Library/Formula/hidapi.rb (hunk 1 above)
Comment (rbarazzutti) [inline Library/Formula/hidapi.rb]: Fine, just removed

File: Library/Formula/hidapi.rb (hunk 3 above)
Comment (rbarazzutti) [inline Library/Formula/hidapi.rb]: done :-)

File: Library/Formula/hidapi.rb (hunk 4 above)
Comment (rbarazzutti) [inline Library/Formula/hidapi.rb]: done :wink:

File: Library/Formula/hidapi.rb (hunk 2 above)
Comment (rbarazzutti) [inline Library/Formula/hidapi.rb]: In a previous patch, I've been told that an explanation why the patch is need would be appreciated.

File: Library/Formula/hidapi.rb (hunk 1 above)
Comment (DomT4) [inline Library/Formula/hidapi.rb]: Thanks! You're welcome to get a second opinion from the maintainers, but we don't have many places where patch authors are written into formulae :).

File: Library/Formula/hidapi.rb (hunk 2 above)
Comment (DomT4) [inline Library/Formula/hidapi.rb]: Oh the patch explanation is fine :), I just meant the blank comment on line 15.

File: Library/Formula/hidapi.rb (hunk 3 above)
Comment (DomT4) [inline Library/Formula/hidapi.rb]: Thanks! And thanks for fixing the double quotes at the top of the formula as well! :+1:

File: Library/Formula/hidapi.rb (hunk 4 above)
Comment (DomT4) [inline Library/Formula/hidapi.rb]: :heart:

**Comment by DomT4** [comment]
One more audit failure and then this'll pass:

```
==> brew audit hidapi							 FAILED
Error: 1 problems in 1 formulae
hidapi:
 * 7: Trailing whitespace was found
```

**Comment by mikemcquaid** [comment]
@DomT4 I'll fix that when I pull.

**Comment by DomT4** [comment]
@mikemcquaid Cool, Thanks!

File: Library/Formula/hidapi.rb (hunk 3 above)
Comment (rbarazzutti) [inline Library/Formula/hidapi.rb]: Thanks for all the hints! :wink:

File: Library/Formula/hidapi.rb (hunk 2 above)
Comment (rbarazzutti) [inline Library/Formula/hidapi.rb]: fine then :+1:  I though that the whole comment was too much verbose :smile:

**Comment by rbarazzutti** [comment]
@mikemcquaid  & @DomT4 I removed the trailing space. It should be okay now.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `hidapi: apply patch to fix IOHidManager.`.

**Pull request closed without merging**